src/plugins/vis_builder/public/application/utils/use/use_persisted_agg_params.test.tsx

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { CreateAggConfigParams } from '../../../../../data/common';
+import { AggGroupNames } from '../../../../../data/common';
+import { Schema } from '../../../../../vis_default_editor/public';
+import { VisBuilderServices } from '../../../types';
+import { createVisBuilderServicesMock } from '../mocks';
+import {
+  AggMapping,
+  getSchemaMapping,
+  updateAggParams,
+  usePersistedAggParams,
+} from './use_persisted_agg_params';
+
+describe('updateAggParams', () => {
+  let oldAggParam: CreateAggConfigParams;
+  let aggMap: Map<string, AggMapping>;
+  let assignNewSchemaType: any;
+
+  beforeEach(() => {
+    oldAggParam = {
+      type: '',
+      enabled: true,
+      schema: '',
+    };
+    aggMap = new Map<string, AggMapping>();
+    aggMap.set('old metric', {
+      currentCount: 0,
+      maxCount: 1,
+      name: 'new metric',
+    });
+    aggMap.set('old bucket', {
+      currentCount: 0,
+      maxCount: 1,
+      name: 'new bucket',
+    });
+    assignNewSchemaType = jest.fn();
+  });
+
+  test('return with the undefined schema if the old schema title does not get a mapping', () => {
+    const newAgg = updateAggParams(oldAggParam, aggMap);
+    expect(newAgg).toEqual(oldAggParam);
+  });
```

Comment:
Im not able to understand the link between the title of the test and the `expect`. What are you testing for in this instance?

The title says that you are expecting the function to return an undefined schema but the test expects the `newAgg` to match the `oldAgg`

Reply:
Yea i wrote it in a confusing way. The test says if the aggregations' schema is already undefined, then after persisting, it will remain undefined, so essentially oldAggParam is equal to newAgg since we do not change the schema or anything else.